ic-payments/src/error.rs

```diff
@@ -0,0 +1,178 @@
+use candid::{CandidType, Deserialize};
+use ic_exports::ic_icrc1::endpoints::TransferError;
+use ic_exports::ic_kit::RejectionCode;
+use ic_helpers::tokens::Tokens128;
+use thiserror::Error;
+
+use crate::BalanceError;
+
+pub type Result<T> = std::result::Result<T, InternalPaymentError>;
+
+/// Reason for a transfer to fail
+#[derive(Debug, PartialEq, CandidType, Deserialize, Error)]
+pub enum TransferFailReason {
+    #[error("token canister does not exist or doesn't follow the ICRC-1 standard")]
+    NotFound,
```

Comment:
MethodNotFound ?

Reply:
This can by either method or the canister itself. And we cannot distinguish these cases except if we try to parse string error message.